python/ray/train/constants.py

```diff
@@ -43,7 +44,9 @@ def _get_ray_train_session_dir() -> str:
 TUNE_CHECKPOINT_ID = "_current_checkpoint_id"
 
 # Deprecated configs can use this value to detect if the user has set it.
-_DEPRECATED_VALUE = "DEPRECATED"
+# This has type Any to allow it to be assigned to any annotated parameter
+# without causing type errors.
+_DEPRECATED_VALUE: Any = "DEPRECATED"
```

Comment:
Instead of changing this, could you change the typehints to take this string as a Literal?

Something like:

```python
deprecated_arg: Union[bool, Literal[_DEPRECATED_VALUE]]
```

Reply:
That will mean that users see a weird union element in the function signature.  It would also mean that passing a string in would pass type checking.  So:
```python
def g(deprecated_arg: Union[bool, Literal[_DEPRECATED_VALUE]]): ...

def f(s: str):
  g(s)  # passes!
```
When at the least `g` should be called with `bool` or nothing.

A better approach would be to make an actual sentinel rather than a string.

Unfortunately, I'm too busy to make a more complex change than this.  I only fixed it last year to silence PyRight back when it was reporting library type errors as user type errors.  Pyright reverted that behavior, so I'm less motivated to fix Ray's code.  But please feel free to close this issue if it's clogging up your review queue.